pkg/manager/member/tikv_member_manager.go

```diff
@@ -587,7 +588,15 @@ func (tkmm *tikvMemberManager) syncTidbClusterStatus(tc *v1alpha1.TidbCluster, s
 		return err
 	}
 
+	pattern, err := regexp.Compile(fmt.Sprintf("%s-tikv-\\d+\\.%s-tikv-peer\\.%s\\.svc\\:\\d+", tc.Name, tc.Name, tc.Namespace))
+	if err != nil {
+		return err
+	}
 	for _, store := range storesInfo.Stores {
+		// make sure that the store belongs to the tikv
```

Comment:
in theory, the external tikv can join the cluster like tiflash, and I guess we should ignore them too, we should explain the reason in detail here, e.g. external tiflash/tikv can join the cluster but they are not under operator management.